### PR TITLE
Fix player does't use the Shellphone (Ocean), Shellphone (Underworld)…

### PR DESCRIPTION
**Pull request by hufang360**
Fix player does't use the Shellphone (Ocean), Shellphone (Underworld) and Shellphone (Spawn).

### CODE REVIEW

**Comment by AxisKriel** [comment]
The changelog entry needs to follow the changelog guidelines (https://github.com/Pryaxis/TShock/blob/general-devel/CHANGELOG.md#how-to-add-a-changelog-entry). I suggest:
```Fixed players being unable to use Shellphone (Ocean), Shellphone (Underworld) and Shellphone (Spawn). (@hufang360)```

Other than that, I tested the code changes and they work, looks good to me 👍

**Review by hakusaro — DISMISSED** [review]

File: TShockAPI/GetDataHandlers.cs [hunk 1]
```diff
@@ -3828,31 +3828,56 @@ void Fail(string tpItem)
...
+						args.Player.SelectedItem.type != ItemID.ShellphoneOcean)
 					{
 						TShock.Log.ConsoleDebug("GetDataHandlers / HandleTeleportationPotion rejected not holding the correct item {0} {1}", args.Player.Name, type);
 						return true;
 					}
 
 					if (!args.Player.HasPermission(Permissions.magicconch))
 					{
-						Fail("the Magic Conch");
+						if (args.Player.ItemInHand.type == ItemID.ShellphoneOcean || args.Player.SelectedItem.type != ItemID.ShellphoneOcean)
```
Comment (drunderscore) [inline TShockAPI/GetDataHandlers.cs]: As stated in the previous PR, the proper check would rather be:
```csharp
args.Player.SelectedItem.type == ItemID.ShellphoneOcean
```

File: TShockAPI/GetDataHandlers.cs [hunk 2]
```diff
@@ -3828,31 +3828,56 @@ void Fail(string tpItem)
...
+						args.Player.SelectedItem.type != ItemID.ShellphoneHell)
 					{
 						TShock.Log.ConsoleDebug("GetDataHandlers / HandleTeleportationPotion rejected not holding the correct item {0} {1}", args.Player.Name, type);
 						return true;
 					}
 
 					if (!args.Player.HasPermission(Permissions.demonconch))
 					{
-						Fail("the Demon Conch");
+						if (args.Player.ItemInHand.type == ItemID.ShellphoneHell || args.Player.SelectedItem.type != ItemID.ShellphoneHell)
```
Comment (drunderscore) [inline TShockAPI/GetDataHandlers.cs]: Similarly to above:
```csharp
args.Player.SelectedItem.type == ItemID.ShellphoneHell
```

**Review by drunderscore — CHANGES_REQUESTED** [review]
Code looks good otherwise and works just fine with these changes.

**Review by drunderscore — APPROVED** [review]
Code LGTM! @hakusaro or someone in Pryaxis will have to start CI manually unfortunately...

**Comment by AxisKriel** [comment]
CI is 👌 
Now just needs @hakusaro seal of approval :)

**Comment by hakusaro** [comment]
@AxisKriel you can review these things too, ya know :p

**Comment by hakusaro** [comment]
Just so we're on the same page, y'all are clear on what this does? This _blocks_ the shellphone if the user doesn't have permission. By default, it would let it through? I'm willing to approve it if y'all can re-confirm that this is the _intent_ of the change.

**Comment by hufang360** [comment]
> 
Yes, that's right.

**Review by hakusaro — APPROVED** [review]

**Comment by hakusaro** [comment]
@hufang360 merged! thank you for your awesome work!

**Comment by hakusaro** [comment]
@AxisKriel fwiw,

> The changelog entry needs to follow the changelog guidelines

This is generally true but it won't stop me from merging a PR if it can be easily corrected and the requestor is not a native English speaker. It's not worth nitpicking/dying on a hill for grammar usually. If it's easy to correct or understand it's nbd!

**Pull request merged**